### PR TITLE
Porting Aggregate, Olap and Partition tests from TincBFV to ICG 

### PR DESCRIPTION
**Pull request by oarap**
@vraghavan78 @xinzweb @hsyuan

### CODE REVIEW

**Comment by hsyuan** [comment]
LGTM.

**Comment by xinzweb** [comment]
Just make sure that works when you run it back-to-back twice.

Maybe wrap those setup stuff inside `--start_ignore` and `--end_ignore`.

```
drop language if exists plpythonu;
create language plpythonu;
```

File: src/test/regress/expected/bfv_aggregate.out [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+   INITCOND = '' --Initialize as an empty string when starting
+);
+-- TEST
+-- cook some stats
+-- start_ignore
+set allow_system_table_mods='DML';
+-- end_ignore
+UPDATE pg_class set reltuples=524592::real, relpages=2708::integer where oid = 'attribute_table'::regclass;
+select count_operator('explain select product_id,concat(\'#attribute_\'||attribute_id::varchar||\':\'||attribute) as attr FROM attribute_table GROUP BY product_id;','HashAggregate');
+WARNING:  nonstandard use of \' in a string literal
```
Comment (hlinnaka) [inline src/test/regress/expected/bfv_aggregate.out]: should fix that query like the hint suggests

File: src/test/regress/sql/bfv_aggregate.sql [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+-- CLEANUP
+-- start_ignore
+drop table if exists t;
+drop function array_sort (ANYARRAY) cascade;
+drop function tfp(anyarray,anyelement) cascade;
+drop function ffp(anyarray) cascade;
+-- end_ignore
+
+-- start_ignore
+-- start_ignore
```
Comment (vraghavan78) [inline src/test/regress/sql/bfv_aggregate.sql]: why twice

**Comment by vraghavan78** [comment]
+1

File: src/test/regress/sql/bfv_aggregate.sql (hunk 2 above)
Comment (oarap) [inline src/test/regress/sql/bfv_aggregate.sql]: It didn't work with a single quote, actually neither of the hint's suggestions worked for this case.

**Comment by xinzweb** [comment]
+1. Btw, how long does these tests run on your box? Thanks.

**Comment by oarap** [comment]
I am not sure if there is a gpdb or orca way of timing it but when i run the specific schedule with time below is the output.

```[regress: oarap{master}]$ time run_installcheck catalog bfv_catalog bfv_index bfv_olap bfv_aggregate bfv_partition
(using postmaster on Unix socket, port 15432)
============== dropping database "regression"         ==============
DROP DATABASE
============== creating database "regression"         ==============
CREATE DATABASE
ALTER DATABASE
============== checking optimizer status              ==============
Optimizer disabled. Using planner answer files
============== running regression test queries        ==============
test catalog              ... ok
test bfv_catalog          ... ok
test bfv_index            ... ok
test bfv_olap             ... ok
test bfv_aggregate        ... ok
test bfv_partition        ... ok

=====================
 All 6 tests passed.
=====================


real	0m13.309s
user	0m2.274s
sys	0m0.769s
[regress: oarap{master}]$ time run_installcheck_with_orca_enabled catalog bfv_catalog bfv_index bfv_olap bfv_aggregate bfv_partition
(using postmaster on Unix socket, port 15432)
============== dropping database "regression"         ==============
DROP DATABASE
============== creating database "regression"         ==============
CREATE DATABASE
ALTER DATABASE
============== checking optimizer status              ==============
Optimizer enabled. Using optimizer answer files whenever possible
============== running regression test queries        ==============
test catalog              ... ok
test bfv_catalog          ... ok
test bfv_index            ... ok
test bfv_olap             ... ok
test bfv_aggregate        ... ok
test bfv_partition        ... ok

=====================
 All 6 tests passed.
=====================


real	0m25.467s
user	0m2.296s
sys	0m0.775s
[regress: oarap{master}]$
```

**Comment by xinzweb** [comment]
Closed using commit: https://github.com/greenplum-db/gpdb/commit/8443c62eba506f4a1e49a803a416811584aa44b0

**Pull request closed without merging**